Bump Java toolchain main version to 17 to match minimal supported Java version

### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

The project can be built right after checkout if a JDK matching the required minimal Java version is installed (Java 17).

### Current Behavior

Cannot build the project. When running the gradle build task. I tried to set the project JDK both to temurin 11 and 17. Neither worked.

### Failure Information (for bugs)

When running build task, I get:
>Could not determine the dependencies of task ':modules:mockk:compileTestKotlinJvm'.
> >Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=11, vendor=any, implementation=vendor-specific} for WINDOWS on x86_64.
>   >No locally installed toolchains match and toolchain download repositories have not been configured.

#### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. checkout project
2. Setup JDK 17
3. run gradle build

#### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* MockK version: current main branch (1.13.12, 5dc7491a46bc374f6d34bc2c5e185b7560518b71)
* OS: Windows 10 x64
* Kotlin version: -
* JDK version: 17
* JUnit version: -
* Type of test: -

#### Failure Logs

Please include any relevant log snippets or files here.

#### Stack trace

```
Could not determine the dependencies of task ':modules:mockk:compileTestKotlinJvm'.
> Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=11, vendor=any, implementation=vendor-specific} for WINDOWS on x86_64.
   > No locally installed toolchains match and toolchain download repositories have not been configured.

* Try:
> Learn more about toolchain auto-detection at https://docs.gradle.org/8.9/userguide/toolchains.html#sec:auto_detection.
> Learn more about toolchain repositories at https://docs.gradle.org/8.9/userguide/toolchains.html#sub:download_repositories.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.9/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Publishing build scan...
https://gradle.com/s/25xprmd6wjtt2
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -10,9 +10,9 @@ kotlin.mpp.stability.nowarn=true
 localrepo=build/maven-local-repo
 ########################
 # version of Java that will be used to build the project
-io_mockk_java_toolchain_main_version=11
+io_mockk_java_toolchain_main_version=17
 # the Java version tests will run against - this is overridden in the GitHub actions
-io_mockk_java_toolchain_test_version=11
+io_mockk_java_toolchain_test_version=17
 ########################
 # Android properties:
 android.useAndroidX=true

